### PR TITLE
Feature/delete confirmation modal

### PR DESCRIPTION
**Pull request by mnunezsa95**
## Describe your changes
- Add markup for delete confirmation modal
- Create a file called "delete-fest.js" in the graphql folder with an API call to delete a fest
- Inside fest-route.jsx, add logic for removing a fest, refetching the fests,
- Inside fest-sidebar.jsx add logic for ensuring only creators of fests can see the delete button and ensure only creators can delete a fest

## Issue ticket number and link
[SLOP 50](https://apiary-edtonomy.atlassian.net/jira/software/projects/SLOP/boards/2?selectedIssue=SLOP-50)
 
## Checklist before requesting a review

- [X] I have performed a self-review of my code
- [ ] If there are changes to components, I have added / updated automated tests
- [ ] I have made any necessary updates to Storybook to accompany these changes
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings

When the user clicks on the "Delete" button on the left sidebar, they will get a popup. They can either confirm or cancel. Cancel will close the modal (nothing happens). Clicking "confirm" will send out a useMutation call and delete the fest from the backend. 

I have tested this by creating a fest on the keystone dashboard, and manually entering that item id at the route /fests/:festId

The item is deleted from Keystone and no longer visible through DBeaver.

### CODE REVIEW

**Comment by SamPage4427** [comment]
Branch should be named in this format: feature/slop-#(ticket number)/delete-confirmation-modal. It should be fine for now but it should include the ticket number. Also don't forget to describe your changes and put the issue ticket number/link when creating your pull request. Use this link as an [example](https://github.com/jahorwitz/slopopedia/pull/34) of how it should look.

**Comment by mnunezsa95** [comment]
> Branch should be named in this format: feature/slop-#(ticket number)/delete-confirmation-modal. It should be fine for now but it should include the ticket number. Also don't forget to describe your changes in the describe changes section and put the issue ticket number/link when creating your pull request. Use this [link](https://github.com/jahorwitz/slopopedia/pull/34) as an example of how it should look.
> 
> 

Thanks! I will make the updates tomorrow morning. As for the branch name is there a way to rename it?

**Comment by SamPage4427** [comment]
The branch name should be fine this time just as a future reference you should add the slop ticket number to it like the example I gave above. You can rename the pull request itself at the top to add the ticket number if you want but the branch should be fine this time.

**Comment by skwisgaarr8-9** [comment]
Seems like the main functionality works here. My only notes would be on the styling within the modal itself. 
image

- I think that the the text `There is no way to undo this action...` should not have the `y-scale`. It looks a little jarring. 
- Also, the cancel button should not have rounded edges; that does not fit with the overall theme of the site. All other buttons and badges have sharp edges.

**Comment by skwisgaarr8-9** [comment]
> As for the branch name is there a way to rename it?

[Renaming a branch](https://stackoverflow.com/questions/6591213/how-can-i-rename-a-local-git-branch)

**Comment by mnunezsa95** [comment]
> Seems like the main functionality works here. My only notes would be on the styling within the modal itself.
> 
> image
> * I think that the the text `There is no way to undo this action...` should not have the `y-scale`. It looks a little jarring.
> * Also, the cancel button should not have rounded edges; that does not fit with the overall theme of the site. All other buttons and badges have sharp edges.

Thanks for pointing this out, I have fixed the two recommendations

Screenshot 2024-02-06 at 7 15 54 AM

**Comment by mnunezsa95** [comment]
> > As for the branch name is there a way to rename it?
> 
> [Renaming a branch](https://stackoverflow.com/questions/6591213/how-can-i-rename-a-local-git-branch)

It seems like that is for branches that have no been pushed to a remote repository. I think this change would require a new pull request. Is it okay, if I use the correct naming moving forward?

**Comment by skwisgaarr8-9** [comment]
> > > As for the branch name is there a way to rename it?
> 
> > 
> 
> > [Renaming a branch](https://stackoverflow.com/questions/6591213/how-can-i-rename-a-local-git-branch)
> 
> 
> 
> It seems like that is for branches that have no been pushed to a remote repository. I think this change would require a new pull request. Is it okay, if I use the correct naming moving forward? 

Yes, like Sam said from next time is fine.

**Comment by skwisgaarr8-9** [comment]
I just noticed that you are not using the `Button` component for either of the buttons in the model. You should be using the `Button` component for both. `Button` has several different variants you can use so take a look and see which variants would be best for each "cancel" and "confirm".

**Comment by mnunezsa95** [comment]
> I just noticed that you are not using the `Button` component for either of the buttons in the model. You should be using the `Button` component for both. `Button` has several different variants you can use so take a look and see which variants would be best for each "cancel" and "confirm".

I realized there was a Button element when working on my new ticket. I just went in and fixed that on this ticket as well! All set

File: src/components/delete-confirmation-modal.jsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+            className="font-arialBold mr-2"
+            onClick={confirmButtonAction}
+          >
+            Confirm
+          </Button>
+          <Button
+            variant="primary"
+            size="sm"
+            className="font-arialBold mr-2"
+            onClick={() => closeModal("")}
```
Comment (skwisgaarr8-9) [inline src/components/delete-confirmation-modal.jsx:27]: `closeModal()` does not take any arguments.

File: src/routes/fest-route/fest-route.jsx [hunk 2]
```diff
@@ -21,6 +32,20 @@ export const FestRoute = () => {
...
+  const [deleteFest, { data, loading, error }] = useMutation(DELETE_FEST, {
+    refetchQueries: [GET_USER_FESTS],
+  });
+
+  const removeFest = () => {
+    if (currentUser.id === festQuery.data.fest.creator.id) {
+      deleteFest({ variables: { where: { id: festId } } });
+      if (loading) return "Submitting...";
+      if (error) return `Submission error! ${error.message}`;
+      closeModal("confirmation");
```
Comment (skwisgaarr8-9) [inline src/routes/fest-route/fest-route.jsx:44]: `closeModal()` does not take any arguments

**Comment by skwisgaarr8-9** [comment]
> > I just noticed that you are not using the `Button` component for either of the buttons in the model. You should be using the `Button` component for both. `Button` has several different variants you can use so take a look and see which variants would be best for each "cancel" and "confirm".
> 
> I realized there was a Button element when working on my new ticket. I just went in and fixed that on this ticket as well! All set

great, styling looks much better now

File: src/components/delete-confirmation-modal.jsx (hunk 1 above)
Comment (mnunezsa95) [reply src/components/delete-confirmation-modal.jsx]: Fixed, oversight on my end

File: src/routes/fest-route/fest-route.jsx (hunk 2 above)
Comment (mnunezsa95) [reply src/routes/fest-route/fest-route.jsx]: Fixed, oversight on my end

**Comment by skwisgaarr8-9** [comment]
Everything looks good here. Moving to Code Approved on Jira

**Comment by mnunezsa95** [comment]
> Everything looks good here. Moving to Code Approved on Jira

Does this mean I can merge into main?

**Comment by SamPage4427** [comment]
> Does this mean I can merge into main?

You don't merge anything. Josh does that

**Comment by mnunezsa95** [comment]
> > Does this mean I can merge into main?
> 
> 
> 
> You don't merge anything. Josh does that when he accepts the review and the changes you've made
> 
> 

Makes sense! Thanks

**Review by jahorwitz — APPROVED** [review]

**Pull request merged**